tests/basic/clocks.rs

```diff
@@ -0,0 +1,415 @@
+use shuttle::sync::{
+    atomic::AtomicBool,
+    mpsc::{channel, sync_channel},
+    Barrier, Condvar, Mutex, RwLock,
+};
+use shuttle::{check_dfs, check_pct, thread};
+use std::collections::HashSet;
+use std::sync::{atomic::Ordering, Arc};
+use test_env_log::test;
+
+pub fn me() -> usize {
+    usize::from(thread::current().id())
+}
+
+// TODO Maybe make this a macro so backtraces are more informative
+pub fn check_clock(f: impl Fn(usize, u32) -> bool) {
+    for (i, &c) in (&thread::my_clock()).into_iter().enumerate() {
+        assert!(f(i, c));
+    }
+}
+
+fn clock_mutex(num_threads: usize) {
```

Comment:
wonder whether this file should have one submodule per sync primitive just to make the organization a bit clearer